multicluster/apis/multicluster/v1alpha1/acnpimport.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=acnpimports,scope=Cluster
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ACNPImport describes an ACNP imported from the leader cluster in a ClusterSet.
+type ACNPImport struct {
```

Comment:
It is created in the leader cluster Common Area or in member cluster? Why we cannot reuse ResourceImport but need another type?

Reply:
It's created in member clusters to denote whether the ACNP can be imported successfully in this particular cluster or not. In the future, the ACNPImport (or ServiceImport for this matter) should be the sole representation of the imported resource in that cluster, meaning that Antrea controller will not create an additional ACNP/Service. By that time, the ACNPImport will have a spec field, and the ACNP controller shall be modified to directly reconcile ACNPImport objects.

Reply:
If it is just for reporting import status, could not we use ResourceImportClusterStatus in the leader cluster?

Reply:
There are two reasons why I feel like ACNPImport is needed in member clusters
1. It provides visibility of ACNP realization status in the member cluster itself, rather than only in common area which individual member clusters do not have access to (unless this is the desired behavior)
2. Ideally I agree that ResourceImport or ResourceExport should also include ACNP realization status for each cluster, but `ResourceImportClusterStatus` is a field that has slight different meaning in other resource's context. Take Service for example: `ResourceImportClusterStatus` will be `succeed` for clusterA if the ResourceExport controller successfully processed Service in clusterA and was able updated ResourceImport for that Service. It does not take into account whether the member cluster controller, in reconciling the ResourceImport, encounters any errors. This means that as of now there's no reporting per cluster status back to the common area for any resource (@luolanzone @aravindakidambi please correct me if I'm wrong). If we are to add that, we should add it for all resources IMO

Reply:
I feel 1) is not very important, as import is more a push model from leader cluster. But I like to understand what other folks think about 2). I thought as it is "import" status, it should be about whether import to a member cluster succeeded or not.

Reply:
Hi @jianjuns @Dyanngg , for item 2, Yang's understanding is correct. In current implementation, the `ResourceImportClusterStatus` is used to indicate if the ResourceExport from a Cluster is converged into ResourceImport successfully. not used for member cluster to report the import status back to leader. 
I personally think the member cluster care more about the status instead of leader cluster.  and the status of a resource should focus on the resource itself.  the leader cluster is a way/role to publish what it has in a ClusterSet. more like a publish-subscribe model. maybe the overall status of imported resources in each member cluster can be achieved via antctl.

Reply:
But that only applies to only a subset of resources that exports from multiple clusters will be merged to one? For example, it does not make any sense for ACNP. And I also feel better to have a way to know about import status from leader cluster API. At least we need the overall import/realization status, like what we support for a ACNP realization in `NetworkPolicyStatus` (https://github.com/antrea-io/antrea/blob/5d9e963bd6d26974860c4f528852ad3187ce15a8/pkg/apis/crd/v1alpha1/types.go#L339)

Reply:
And should not that status of "successfully created/updated ResourceImport" in ResourceExport Status? It is about processing/reconciliation of ResourceExport right?

Reply:
@jianjuns  after review existing codes for ResourceExport/ResourceImport status, controller will actually update the ResourceExport's status to reflect the coverage status. so I suppose we can simply remove the status in ResourceImport if we'd like to reuse it for member cluster to report importing status back. since @Dyanngg is trying to use event to report the status back, so I will keep it for now until we have a final decision.

Reply:
Sounds good. I feel for this PR, probably we can remove the ACNPImport and Status update part.

Reply:
@luolanzone @jianjuns I've removed ACNPImport and added K8s events for reporting ACNP creation status in the latest commit.